### PR TITLE
Add `science complete` for shell completion.

### LINKED ISSUE #20: The `science` binary supports shell completion, document this and make it easier to setup.
Currently you'd have to know science is Python and uses `click` to figure out to do (for bash):
```bash
eval "$(_SCIENCE_COMPLETE=bash_source science)"
```

The science binary should probably have a sub-command to generate its completion scripts / install them.

### PR DESCRIPTION
**Pull request by jsirois**
This both wraps up generation of the completion script and documents
the feature.

Closes #20

### CODE REVIEW

**Comment by jsirois** [comment]
I tested this out for Windows and it does correctly detect (Git) bash, but does not generate the completion script due to: https://github.com/pallets/click/issues/2461

File: nox-support/doc-reqs.windows-amd64.lock.txt [hunk 1]
```diff
@@ -10,16 +10,8 @@ sphinx==6.2.1 \
 sphinx-click==4.4.0 \
   --hash=sha256:2821c10a68fc9ee6ce7c92fad26540d8d8c8f45e6d7258f0e4fb7529ae8fab49 \
   --hash=sha256:cc67692bd28f482c7f01531c61b64e9d2f069bfcf3d24cbbb51d4a84a749fa48
-
-# N.B.: The 21c3f448d17224246e36a5db678e98b9261a1909604c160cd12f45fa1e7eb695 hash
```
Comment (jsirois) [inline nox-support/doc-reqs.windows-amd64.lock.txt:14]: Heh - added the note and still failed.

Comment (jsirois) [reply nox-support/doc-reqs.windows-amd64.lock.txt]: Fixed.

**Comment by jsirois** [comment]
I tested this out in all 3 shells. The zsh and fish completion is especially nice with the help summary lines being displayed alongside the sub-command or option.

The rendered CLI docs look like:
Screenshot 2023-06-09 095753

File: science/commands/complete.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+        scie_process = python_process.parent()
+        if not scie_process:
+            return None
+
+        shell_process = scie_process.parent()
+        if not shell_process:
+            return None
+
+        parent_exe_name = scie_process.name()
```
Comment (jsirois) [inline science/commands/complete.py:31]: That's a 2x bug.

Comment (jsirois) [reply science/commands/complete.py]: Fixed.

**Review by benjyw — APPROVED** [review]

**Review by sureshjoshi — APPROVED** [review]
Just trying to catch up with recent changes.

Is the README/documentation for calling `eval $(...)` intended to be in this PR?

**Comment by jsirois** [comment]
README will ~die. I'm getting nearly all the real doc content in the doc strings -> the doc site -> embedded in the binary + (for CLI help like this) --help.

**Comment by jsirois** [comment]
Just clicked the publish button on main; so here on the doc site: https://science.scie.app/cli.html#science-complete

**Pull request merged**